packages/@lwc/template-compiler/src/shared/types.ts

```diff
@@ -170,6 +170,10 @@ export interface Element extends AbstractBaseElement {
     type: 'Element';
 }
 
+export interface External extends AbstractBaseElement {
```

Comment:
Just for my understanding, what's the difference between having this `External` as BaseElement vs. using `ExternalDirective extends Directive<'External'>`?

Reply:
Maybe we should rename it but `External` is the type for the element with the directive and `ExternalDirective` is the type of the directive itself (i.e., `lwc:external`).

Reply:
Ok, but we don't really need `ExternalDirective` as one of `ElementDirective` type because it is not used anywhere and only `ElementDirectiveName.External` would be enough?

Reply:
@rui-rayqiu I initially defined the `ExternalDirective` type because I was interacting with it by adding it to the AST, but after I introduced the `ExternalComponent` type instead, I ended up not including the directive in the AST.